### ISSUE TITLE
Multiprovisioner persistent volume support

### ISSUE THREAD

**Issue by cezarsa** [tsuru/tsuru#1599]
The objective of this proposal is allowing tsuru apps to use external volumes and mount them on apps running on tsuru. The concepts of `volume-plan`, `volume` and `volume binding` would be added to tsuru.

### `volume-plan`

Volume plans would be statically defined in `tsuru.conf` file and they describe how a new volume can be created in each provisioner. Example:

```
...
volume-plans:
  - name: nfs
    swarm:
       driver: local
       opt:
         - type=nfs
    kubernetes:
      plugin: nfs
...
```

And it should also be possible to use storage classes with kubernetes and external drivers with docker swarm:

```
...
volume-plans:
  - name: ebs
    swarm:
       driver: rexray/ebs
    kubernetes:
      storage-class: my-ebs-storage-class
...
```

### `volume`

Volumes would be created using tsuru api and require an existing `volume-plan`. The creation of the volume would be delegated to the provisioner. The basic command for adding volumes would be:

```
tsuru volume-create <name> --provisioner <provisioner> --team <team> \
  --plan <volume-plan-name> [--opt key=value]...
```

#### With kubernetes provisioner

Creating a volume with a plan that has no storage-class defined will cause tsuru to manually create one `PersistentVolume` using the `plugin` specified in the plan with the `opt` received in the command line. Also, one `PersistentVolumeClaim` would be created and bound to the `PersistentVolume`.

If the plan specifies a `storage-class` instead of a `plugin` only the `PersistentVolumeClaim` will be created using the specified storage-class.

#### With swarm provisioner

A new volume would be created (i.e. `docker volume create`) using the `driver` informed in the plan and the volume `opt` would be a merge between the plan `opt` and command line `opt`.

#### Examples

```
tsuru volume-create mynfskube --provisioner kubernetes --team t1 \
    --plan nfs --opt path=/exports/dir --opt server=192.168.50.1 --opt capacity=50Gi --opt access-modes=RWX
```
(The `capacity` and `access-modes` opts would be handled as a special case in the provisioner, setting them to the correct fields in the `PersistentVolume`)

```
tsuru volume-create mynfsswarm --provisioner swarm --team t1 \
    --plan nfs --opt o=addr=192.168.50.1,rw --opt device=:/exports/dir
```

#TODO volume bind

**Comment by andrestc** [comment]
As discussed, I think `volume-create` should receive the `pool` as a parameter, instead of `provisioner` to account for the cases where we have multiple kubernetes clusters. Only apps from that same pool should be able to bind to that volume (even if two pools share the same kubernetes cluster).

**Comment by andrestc** [comment]
Volume support was added to both swarm and kubernetes on an experimental version. Available on tsuru 1.4.
